src/crawlee/_request.py

```diff
@@ -119,7 +166,7 @@ class BaseRequestData(BaseModel):
     method: HttpMethod = 'GET'
     """HTTP request method."""
 
-    headers: Annotated[dict[str, str], Field(default_factory=dict)] = {}
+    headers: Annotated[HttpHeaders, Field(default_factory=HttpHeaders())] = HttpHeaders()
```

Comment:
Suspicious default factory, do we really need both that and a regular default?

Also, doesn't this cause `Request` to serialize to `{..., "headers": {"headers": {...}}, ...}` now? Wouldn't a `RootModel` be more appropriate? :slightly_smiling_face:

Reply:
> Suspicious default factory, do we really need both that and a regular default?

yeah, mypy

> Also, doesn't this cause Request to serialize to {..., "headers": {"headers": {...}}, ...} now? Wouldn't a RootModel be more appropriate? 🙂

Thanks!